harness/determined/pytorch/_pytorch_context.py

```diff
@@ -459,33 +473,31 @@ def configure_apex_amp(
         if not enabled or not self.env.managed_training:
             return models, optimizers
 
-        if self._scaler is not None:
-            check.false(self._scaler.is_enabled(), "Do not mix APEX with PyTorch AMP")
-
-        check.false(self._use_apex, "Please only call configure_apex_amp once.")
-        if self.distributed.size > 1:
-            check.eq(
-                num_losses,
-                1,
-                "When using parallel/distributed training, "
-                "Determined only supports configure_apex_amp with num_losses = 1",
-            )
+        if self._scaler is not None and self._scaler.is_enabled():
+            raise det.errors.InvalidExperimentException("Do not mix APEX with PyTorch AMP.")
 
-        self._use_apex = True
+        if self._use_apex:
+            raise det.errors.InvalidExperimentException("Please only call configure_apex_amp once.")
 
         if self.distributed.size > 1:
-            check.eq(
-                self._aggregation_frequency,
-                1,
-                "Mixed precision training (AMP) is not supported with "
-                "distributed training and "
-                "aggregation frequency > 1.",
+            if num_losses != 1:
+                raise det.errors.InvalidExperimentException(
+                    "When using parallel/distributed training, "
+                    "Determined only supports configure_apex_amp with num_losses = 1.",
+                )
+            if self._aggregation_frequency > 1:
+                raise det.errors.InvalidExperimentException(
+                    "Mixed precision training (AMP) is not supported with "
+                    "distributed training and "
+                    "aggregation frequency > 1.",
+                )
+
+        if not torch.cuda.is_available():
+            raise det.errors.InvalidExperimentException(
+                "Mixed precision training (AMP) is supported only on GPU slots.",
```

Comment:
as before, please modify the text to include that the `configure_apex_amp` is what triggered the error.

Reply:
Done.